joint_trajectory_controller/include/joint_trajectory_controller/tolerances.hpp

```diff
@@ -0,0 +1,320 @@
+//     documentation and/or other materials provided with the distribution.
+//   * Neither the name of PAL Robotics S.L. nor the names of its
+//     contributors may be used to endorse or promote products derived from
+//     this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//////////////////////////////////////////////////////////////////////////////
+
+/// \author Adolfo Rodriguez Tsouroukdissian
+
+#ifndef JOINT_TRAJECTORY_CONTROLLER_TOLERANCES_H
+#define JOINT_TRAJECTORY_CONTROLLER_TOLERANCES_H
+
+// C++ standard
+#include <cassert>
+#include <cmath>
+#include <string>
+#include <vector>
+
+// ROS
+#include <rclcpp/rclcpp.hpp>
+
+// ROS messages
+#include "control_msgs/action/follow_joint_trajectory.hpp"
+
+namespace joint_trajectory_controller
+{
+
+/**
+ * \brief Trajectory state tolerances for position, velocity and acceleration variables.
+ *
+ * A tolerance value of zero means that no tolerance will be applied for that variable.
+ */
+struct StateTolerances
+{
+  StateTolerances(double position_tolerance     = 0.0,
+                  double velocity_tolerance     = 0.0,
+                  double acceleration_tolerance = 0.0)
+    : position(position_tolerance),
+      velocity(velocity_tolerance),
+      acceleration(acceleration_tolerance)
+  {}
+
+  double position;
+  double velocity;
+  double acceleration;
+};
+
+/**
+ * \brief Trajectory segment tolerances.
+ */
+struct SegmentTolerances
+{
+  SegmentTolerances(const typename std::vector<StateTolerances>::size_type& size = 0)
+    : state_tolerance(size, 0.0),
+      goal_state_tolerance(size, 0.0),
+      goal_time_tolerance(0.0)
+  {}
+
+  /** State tolerances that appply during segment execution. */
+  std::vector<StateTolerances> state_tolerance;
+
+  /** State tolerances that apply for the goal state only.*/
+  std::vector<StateTolerances> goal_state_tolerance;
+
+  /** Extra time after the segment end time allowed to reach the goal state tolerances. */
+  double goal_time_tolerance;
+};
+
+/**
+ * \brief Trajectory segment tolerances per Joint
+ */
+struct SegmentTolerancesPerJoint
+{
+  SegmentTolerancesPerJoint()
+    : state_tolerance(0.0),
```

Comment:
we no longer need to specify the initialization of these values in the default constructor, please move them into the body